L1Trigger/TrackFindingTMTT/src/Histos.cc

```diff
@@ -0,0 +1,1776 @@
+
+  TFileDirectory Histos::bookEtaPhiSectors() {
+    TFileDirectory inputDir = fs_->mkdir("CheckSectors");
+
+    // Check if stubs excessively duplicated between overlapping sectors.
+    hisNumEtaSecsPerStub_ =
+        inputDir.make<TH1F>("NumEtaSecPerStub", "; Number of #eta sectors each stub appears in", 20, -0.5, 19.5);
+    hisNumPhiSecsPerStub_ =
+        inputDir.make<TH1F>("NumPhiSecPerStub", "; Number of #phi sectors each stub appears in", 20, -0.5, 19.5);
+
+    // Count stubs per (eta,phi) sector.
+    hisNumStubsPerSec_ = inputDir.make<TH1F>("NumStubsPerSec", "; Number of stubs per sector", 250, -0.5, 249.5);
+
+    return inputDir;
+  }
+
+  //=== Fill histograms checking if (eta,phi) sector definition choices are good.
+
+  void Histos::fillEtaPhiSectors(const InputData& inputData, const matrix<unique_ptr<Sector>>& mSectors) {
+    // Allow only one thread to run this function at a time
+    std::lock_guard<std::mutex> myGuard(myMutex[1]);
+
+    const list<const Stub*>& vStubs = inputData.stubsConst();
+    //const list<TP>& vTPs = inputData.getTPs();
+
+    //=== Loop over all stubs, counting how many sectors each one appears in.
+
+    for (const Stub* stub : vStubs) {
+      // Number of (eta,phi), phi & eta sectors containing this stub.
+      unsigned int nEtaSecs = 0;
+      unsigned int nPhiSecs = 0;
+
+      // Loop over (eta, phi) sectors.
+      for (unsigned int iPhiSec = 0; iPhiSec < numPhiSectors_; iPhiSec++) {
+        for (unsigned int iEtaReg = 0; iEtaReg < numEtaRegions_; iEtaReg++) {
+          const Sector* sector = mSectors(iPhiSec, iEtaReg).get();
+
+          // Check if sector contains stub stub, and if so count it.
+          // Take care to just use one eta (phi) typical region when counting phi (eta) sectors.
+          if (iPhiSec == 0 && sector->insideEta(stub))
+            nEtaSecs++;
+          if (iEtaReg == 0 && sector->insidePhi(stub))
+            nPhiSecs++;
+        }
+      }
+
+      // Plot number of sectors each stub appears in.
+      hisNumEtaSecsPerStub_->Fill(nEtaSecs);
+      hisNumPhiSecsPerStub_->Fill(nPhiSecs);
+    }
+
+    //=== Loop over all sectors, counting the stubs in each one.
+    for (unsigned int iEtaReg = 0; iEtaReg < numEtaRegions_; iEtaReg++) {
+      for (unsigned int iPhiSec = 0; iPhiSec < numPhiSectors_; iPhiSec++) {
+        const Sector* sector = mSectors(iPhiSec, iEtaReg).get();
+
+        unsigned int nStubs = 0;
+        for (const Stub* stub : vStubs) {
+          if (sector->inside(stub))
+            nStubs++;
+        }
+        hisNumStubsPerSec_->Fill(nStubs);
+      }
+    }
+  }
+
+  //=== Book histograms checking filling of r-phi HT array.
+
+  TFileDirectory Histos::bookRphiHT() {
+    TFileDirectory inputDir = fs_->mkdir("HTrphi");
+
+    return inputDir;
+  }
+
+  //=== Fill histograms checking filling of r-phi HT array.
+
+  void Histos::fillRphiHT(const matrix<unique_ptr<HTrphi>>& mHtRphis) {
```

Comment:
this function now doesn't do anything

Reply:
We deleted about half the histograms booked by the Histos class in the initial PR, so as to simplify things. In a few places, this left empty functions. But we prefer not to delete them, so that L1 track people wishing to add private histos for their studies have the correct places to insert them.